### PR TITLE
07 start

### PR DESCRIPTION
**Pull request by adamkohler**
made it to last part of exercise.  LuckList page does not show any data.  Could not figure out last part.

### CODE REVIEW

File: LuckySpin/Controllers/SpinnerController.cs [hunk 1]
```diff
@@ -57,29 +58,29 @@ public IActionResult Index(Player player)
 
             //TODO: pass the player.Id to the SpinIt action
             //      (remember, you have to pass it as an object property so use the 'new { }' syntax)
-            return RedirectToAction("SpinIt");
+            return RedirectToAction("SpinIt", new { player.Id });
```
Comment (Bansenauer-Cascadia) [inline LuckySpin/Controllers/SpinnerController.cs]: Here you successfully pass an object, but it is good to name the property using the syntax, new {name = value} like
 new { id=player.Id }

File: LuckySpin/Controllers/SpinnerController.cs [hunk 2]
```diff
@@ -91,7 +92,8 @@ public IActionResult Index(Player player)
                 ViewBag.Display = "none";
             //TODO Assign a ViewBag.PlayerId item used to assigns a link its route_id in SpinIt View
             //      (see the <a href> for "Current Balance" in the SpinIt.cshtml file)
-
+            ViewBag.PlayerID = id;
+            //Confused on what to do here.
```
Comment (Bansenauer-Cascadia) [inline LuckySpin/Controllers/SpinnerController.cs]: The given code uses the Spin property of the _dbc directly. Run the code as is and review the database record in the “spin” table that it creates.

Then change the code to use the currentPlayer.Spins property to add the new spin, and see what difference that makes in the database. Pay special attention to the value of the field for the playerId in the database.

File: LuckySpin/Controllers/SpinnerController.cs [hunk 3]
```diff
@@ -105,13 +107,12 @@ public IActionResult Index(Player player)
          * ListSpins Action
          **/
 
-         public IActionResult LuckList(int id)
+        public IActionResult LuckList(int id)
```
Comment (Bansenauer-Cascadia) [inline LuckySpin/Controllers/SpinnerController.cs]: This method looks fine.

File: LuckySpin/Controllers/SpinnerController.cs (hunk 2 above)
Comment (Bansenauer-Cascadia) [reply LuckySpin/Controllers/SpinnerController.cs]: Until you make that change, the new spins will not be associated with the player. That is why your list is coming up empty.

**Comment by Bansenauer-Cascadia** [comment]
Adam,

You are very close here, just a couple of missing pieces. See my code comments in my code review for  pull request “files changed” section

**Comment by adamkohler** [comment]
I am unsure if I did the new pull request correctly or if you can see the updated code.  This took longer then I wanted but I do want to make sure it is correct.  Thanks

File: LuckySpin/Controllers/SpinnerController.cs [hunk 4]
```diff
@@ -91,11 +92,14 @@ public IActionResult Index(Player player)
...
-
+            ViewBag.PlayerID = id;
+            //Confused on what to do here.
 
             //TODO Compare DB records when adding a generic Spin, as shown below, 
             //     with adding a new Spin to the current player's Spins list
-            _dbc.Spins.Add(new Spin { IsWinning = spinVM.IsWinning });
+            //_dbc.Spins.Add(new Spin { IsWinning = spinVM.IsWinning });
+            var spin = new Spin();
+            currentPlayer.Spins.Add(spin);
```
Comment (Bansenauer-Cascadia) [inline LuckySpin/Controllers/SpinnerController.cs]: This is closer - but "spin" has no properties, so it is not being stored in the database. Combine your earlier code with the use of the currentPlayer Spin property, like so,

currentPlayer.Spins.Add(new Spin { IsWinning = spinVM.IsWinning} );

**Pull request closed without merging**